### PR TITLE
ICU-22510 Fix Calendar API to ignore or reject invalid field.

### PR DESCRIPTION
**Pull request by FrankYFTang**
Add field check for Calendar API. Either ignore invalid field or return with status code set to .U_ILLEGAL_ARGUMENT_ERROR 

##### Checklist

- [X] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22510
- [X] Required: The PR title must be prefixed with a JIRA Issue number. 
- [X] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [X] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [X] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Review by richgillam — DISMISSED** [review]

**Review by richgillam — APPROVED** [review]
LGTM

File: icu4c/source/i18n/calendar.cpp [hunk 1]
```diff
@@ -1192,6 +1192,11 @@ Calendar::setTimeInMillis( double millis, UErrorCode& status ) {
 int32_t
 Calendar::get(UCalendarDateFields field, UErrorCode& status) const
 {
+    if (U_FAILURE(status)) return 0;
```
Comment (markusicu) [inline icu4c/source/i18n/calendar.cpp:1195]: Please always use `{ ... }` for if-bodies, even single-liners.

**Review by markusicu — APPROVED** [review]
tnx

**Comment by FrankYFTang** [comment]
PTAL

**Review by richgillam — APPROVED** [review]
Still LGTM.

**Review by markusicu — APPROVED** [review]

**Pull request merged**